adapters/limelightDigital/limelightDigital.go

```diff
@@ -0,0 +1,172 @@
+
+		requestJSON, err := json.Marshal(requestCopy)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requestData := &adapters.RequestData{
+			Method: "POST",
+			Uri:    url,
+			Body:   requestJSON,
+		}
+		requests = append(requests, requestData)
+	}
+	return requests, errors
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	if len(responseData.Body) == 0 {
+		return nil, nil
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+	for _, seatBid := range response.SeatBid {
+		for i, bid := range seatBid.Bid {
+			b := &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: getMediaTypeForBid(bid.ImpID, request.Imp),
+			}
+			bidResponse.Bids = append(bidResponse.Bids, b)
+		}
+	}
+	return bidResponse, nil
+}
+
+func getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ImpExtLimelightDigital, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "ext.bidder is not provided",
+		}
+	}
+	var limelightDigitalExt openrtb_ext.ImpExtLimelightDigital
+	if err := json.Unmarshal(bidderExt.Bidder, &limelightDigitalExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "ext.bidder is not provided",
+		}
+	}
+	imp.Ext = nil
+	return &limelightDigitalExt, nil
+}
+
+func (a *adapter) buildEndpointURL(params *openrtb_ext.ImpExtLimelightDigital) (string, error) {
+	endpointParams := macros.EndpointTemplateParams{
+		Host:        params.Host,
+		PublisherID: params.PublisherID.String(),
+	}
+	return macros.ResolveMacros(a.endpointTemplate, endpointParams)
+}
+
+func getMediaTypeForBid(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Banner != nil {
+				return openrtb_ext.BidTypeBanner
+			} else if imp.Video != nil {
+				return openrtb_ext.BidTypeVideo
+			} else if imp.Audio != nil {
+				return openrtb_ext.BidTypeAudio
+			} else if imp.Native != nil {
+				return openrtb_ext.BidTypeNative
+			}
+			return openrtb_ext.BidTypeBanner
```

Comment:
Could you please help me understand why the BidTypeBanner is returned in this case as well on line 171?

Reply:
Thank you so much for your review It really looks like an opportunity for improvement. I improve BidType checking in https://github.com/prebid/prebid-server/pull/2539/commits/a8efcf771fdf43622499d870cd6f88a7e92c9bf2 and create a test which checks this case

Reply:
I renamed test in https://github.com/prebid/prebid-server/pull/2539/commits/fb67dbd204813d43f18ae5a662b523d1e8731fd3 commit

Reply:
Good point. With new changes L174 can be fully removed or modified to `break`
Also please add test for `native` media type

Reply:
@VeronikaSolovei9 I improved unknown media type filtering in commit  https://github.com/prebid/prebid-server/pull/2539/commits/0564ec79541d9b339727fcfe1f29aceb5e150698 commit and added native media type test in https://github.com/prebid/prebid-server/pull/2539/commits/25733951529ac9f1415931d326d3955a93978729 commit